File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/core/Properties.java
package app.zxtune.core;

/**
 * Poperties 'namespace'
 */
public final class Properties {

  /**
   * Prefix for all properties
   */
  public static final String PREFIX = "zxtune.";

  /**
   * Sound properties 'namespace'
   */
  public static final class Sound {

    static final String PREFIX = Properties.PREFIX + "sound.";

    /**
     * Loop mode
     */
    public static final String LOOPED = PREFIX + "looped";
  }

  /**
   * Core properties 'namespace'
   */
  static final class Core {

    static final String PREFIX = Properties.PREFIX + "core.";

    /**
     * AY/YM properties 'namespace'
     */
    static final class Aym {

      static final String PREFIX = Core.PREFIX + "aym.";

      public static final String INTERPOLATION = PREFIX + "interpolation";
    }
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/core/Properties.kt
package app.zxtune.core

/**
 * Poperties 'namespace'
 */
object Properties {
    /**
     * Prefix for all properties
     */
    const val PREFIX = "zxtune."

    object Core {
        private const val PREFIX = Properties.PREFIX + "core."

        /**
         * Channels muting mask (if supported, @see ModuleAttributes.CHANNELS_NAMES)
         */
        const val CHANNELS_MASK = PREFIX + "channels_mask"
    }

    /**
     * Sound properties 'namespace'
     */
    object Sound {
        private const val PREFIX = Properties.PREFIX + "sound."

        /**
         * Loop mode
         */
        const val LOOPED = PREFIX + "looped"
    }
}
